File: care_well_app/lib/domain/entities/health/ficha_salud_enfermedad.dart
import 'package:care_well_app/domain/entities/entities.dart';

class FichaSaludEnfermedad extends BaseEntity {
  final String nombre;
  final bool vigente;
  final String? observacion;

  const FichaSaludEnfermedad({
    super.id = 0,
    required this.nombre,
    required this.vigente,
    this.observacion,
  });

  @override
  FichaSaludEnfermedad copyWith({
    int? id,
    String? nombre,
    bool? vigente,
    String? observacion,
  }) {
    return FichaSaludEnfermedad(
      id: id ?? this.id,
      nombre: nombre ?? this.nombre,
      vigente: vigente ?? this.vigente,
      observacion: observacion,
    );
  }
}

File: care_well_backend/CareWell/CareWell.Domain/Salud/FichaSaludEnfermedad.cs
using CareWell.Domain.ValueObjects.Salud;
using CareWell.Global.Exceptions;
using CareWell.Global.Mensajes;

namespace CareWell.Domain.Salud
{
    public class FichaSaludEnfermedad : BaseEntity
    {
        public virtual FichaSalud FichaSalud { get; private set; }

        public virtual string Nombre { get; private set; }

        public virtual bool Vigente { get; private set; }

        public virtual string? Observacion { get; private set; }

        public virtual void Registrar(AgregarEnfermedad agregarEnfermedad, FichaSalud fichaSalud)
        {
            if (string.IsNullOrEmpty(agregarEnfermedad.Nombre))
                throw new ValidacionDominioException(Mensajes.NombreEnfermedadRequerido);

            this.FichaSalud = fichaSalud;
            this.Nombre = agregarEnfermedad.Nombre;
            this.Vigente = agregarEnfermedad.Vigente;
            this.Observacion = agregarEnfermedad.Observacion;
        }
    }
}
